common/endpoints/endpoints.go

```diff
@@ -8,35 +8,48 @@ import (
 	"io"
 	"log"
 	"net/http"
+	"os"
+	"strings"
+	"sync"
 	"time"
 
 	"github.com/scootdev/scoot/common/stats"
+	"github.com/scootdev/scoot/os/temp"
 )
 
-func NewTwitterServer(addr string, stats stats.StatsReceiver) *TwitterServer {
+// Returns an http handler at 'addr' which can retrieved with 'uri' if specified. Also serves static files from tmpDir if specified.
+func NewTwitterServer(addr, uri string, stats stats.StatsReceiver, tmpDir *temp.TempDir) *TwitterServer {
 	return &TwitterServer{
-		Addr:  addr,
-		Stats: stats,
+		Addr:            addr,
+		Stats:           stats,
+		TmpDir:          tmpDir,
+		ResourceHandler: NewResourceHandler(uri, 2500*time.Millisecond),
 	}
 }
 
 // A stats receiver that provides HTTP access for metric scraping with
 // Twitter-style endpoints.
 type TwitterServer struct {
-	Addr  string
-	Stats stats.StatsReceiver
+	Addr            string
+	Stats           stats.StatsReceiver
+	TmpDir          *temp.TempDir
+	ResourceHandler *ResourceHandler
 }
 
 func (s *TwitterServer) Serve() error {
-	mux := http.NewServeMux()
-	mux.HandleFunc("/", healthHandler)
-	mux.HandleFunc("/admin/metrics.json", s.statsHandler)
-	server := &http.Server{
-		Addr:    s.Addr,
-		Handler: mux,
+	http.HandleFunc("/", helpHandler)
```

Comment:
Err, http.HandleFunc is a global... Could we keep using a specific one?

E.g., for hackweek, I'm adding a second http servemux into a server and it'd be nice to not have to reason about this.